### DISCUSSION TITLE
Layout attribute on dynamically nested components

### DISCUSSION THREAD

**Discussion by benjibee** [livewire/livewire#8654] [1. Help]
Using the layout attribute on the `render()` method works as expected, but the same attribute on the `render()` method of a nested component is simply not rendered.

Parent component (works as expected)
```php
// app/Livewire/Survey/CreateSurvey.php

#[Layout('layouts.app')]
public function render(): View
{
    return view('livewire.survey.create-survey');
}
```

Dynamic nested components in a loop
```blade
// resources/views/livewire/survey/create-survey.php

@foreach($questions as $question)
    <livewire:dynamic-component
        :is="$question->getComponentPath()"
        :key="$loop->index"
        :$question
    />
@endforeach
```

Nested component (Name component)
```php
// app/Livewire/Survey/Questions/Name.php

#[Layout('livewire.survey.questions.index')]
public function render(): View
{
    return view('livewire.survey.questions.name');
}
```

I would expect the parent component to be wrapped in the passed layout attribute AND for the nested component to be wrapped in its provided component as well, but this not what I'm seeing.

**Comment by lrljoe** [comment]
The Layout parameter/attribute is for a full-page Livewire component, and is for a **top level layout**.

It is **not** for child components to select a top-level layout to render themselves in.

For this, you should be using standard blade components!

Otherwise you'll end up with
```
<html>
  <head>
   // Head Stuff
  </head>
  <body>
     // Default Stuff
    <html>
      <head>
       // Other Head Stuff
      </head>    
     <body>
     // Some Body
     </body>
    </html>
    <html>
      <head>
       // Other Head Stuff
      </head>    
     <body>
     // Some Body
     </body>
    </html>
    <html>
      <head>
       // Other Head Stuff
      </head>    
     <body>
     // Some Body
     </body>
    </html>
    <html>
      <head>
       // Other Head Stuff
      </head>    
     <body>
     // Some Body
     </body>
    </html>
</body>
</html>
```
Which just doesn't make any sense to a browser at all, so Livewire/Laravel correctly stop the behaviour.
